src/main/java/uk/gov/hmcts/reform/userprofileapi/service/impl/FeatureToggleServiceImpl.java

```diff
@@ -0,0 +1,55 @@
+package uk.gov.hmcts.reform.userprofileapi.service.impl;
+
+import com.launchdarkly.sdk.LDUser;
+import com.launchdarkly.sdk.server.LDClient;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.userprofileapi.service.FeatureToggleService;
+
+import javax.annotation.PostConstruct;
+import java.util.HashMap;
+import java.util.Map;
+
+@Service
+public class FeatureToggleServiceImpl implements FeatureToggleService {
+
+    @Autowired
+    private final LDClient ldClient;
+
+    @Value("${launchdarkly.sdk.environment}")
+    private String environment;
+
+    private final String userName;
+
+    private Map<String, String> launchDarklyMap;
+
+    @Autowired
+    public FeatureToggleServiceImpl(LDClient ldClient, @Value("${launchdarkly.sdk.user}") String userName) {
+        this.ldClient = ldClient;
+        this.userName = userName;
+    }
+
+    @PostConstruct
+    public void mapServiceToFlag() {
+        launchDarklyMap = new HashMap<>();
+        launchDarklyMap.put("UserProfileController.deleteUserProfileByIdOrEmailPattern",
+                "up-delete-by-id-or-email-pattern");
```

Comment:
Please verify flag name with Shashank

Reply:
@shashank-rastogi has confirmed the flag name to be `delete-user-by-id-or-email-pattern`, change made here: https://github.com/hmcts/rd-user-profile-api/pull/631/commits/dc8401546b54112d1f427da53ee4204c78eb8a92